### ISSUE TITLE
The database connection cannot be released without using the slice key when adding or querying data

### ISSUE THREAD

**Issue by 18511699755** [apache/shardingsphere#6322]
When adding or querying data, the database connection cannot be released without using the slicing key, and sql execution also fails

**Comment by kimmking** [comment]
can you show more details?

**Comment by 18511699755** [comment]
$ curl 127.0.0.1:8085/t
The command will be blocked after execution

There are 256 tables, and there will be 256 Actual SQL, but if there is no SQL execution result, it will be reported that the database link is insufficient if CURL is carried out many times

**Comment by kimmking** [comment]
details such as config rules, logic sql and actual sql .

**Comment by 18511699755** [comment]
I found the reason.
spring.shardingsphere.props.max.connections.size.per.query=32
spring.shardingsphere.datasource.demo.max-active= 20
The two configurations are different. The former is larger than the latter. I have 256 tables here. When the shard key is not hit, all the tables will be scanned. According to the first configuration, 32 database connections will be created, but the database I configured here There are only 20 connections, and the creation process is synchronous, resulting in application process blocking.


It is recommended that when this happens, the framework first judges the size of these two parameters and then selects the smallest one.

**Comment by 18511699755** [comment]
> details such as config rules, logic sql and actual sql .

I found the reason.
spring.shardingsphere.props.max.connections.size.per.query=32
spring.shardingsphere.datasource.demo.max-active= 20
The two configurations are different. The former is larger than the latter. I have 256 tables here. When the shard key is not hit, all the tables will be scanned. According to the first configuration, 32 database connections will be created, but the database I configured here There are only 20 connections, and the creation process is synchronous, resulting in application process blocking.

It is recommended that when this happens, the framework first judges the size of these two parameters and then selects the smallest one.
